packages/tailwindcss/src/index.ts

```diff
@@ -580,7 +581,7 @@ export async function compile(
   // resulted in a generated AST Node. All the other `rawCandidates` are invalid
   // and should be ignored.
   let allValidCandidates = new Set<string>()
-  let compiledCss = features !== Features.None ? toCss(ast) : css
+  let compiled = features !== Features.None ? optimizeAst(ast) : input
```

Comment:
I think we can actually get rid of this `optimizeAst()` call here. There are two scenarios to consider:

1. A subsequent `build()` is called with no changes: This is the case where we currently safe one computation. In that case we can also move the `optimizeAst()` inot the `build()` function though and add something like (`if (!compiled) features !== Features.None ? optimizeAst(ast) : input`) there inline
2. A subsequent `build()` is called **with changes**: I'd say this happens very often since the first build probably more often than not does require candidates which will require a rebuild. In this case we're currently computing the initial `optimizeAst` pass for no reason since it's going to be replaced soon anyway.

This is possible since the `compileAst()` function does never return the compiled CSS, you can only do so by calling `build()`.